### PR TITLE
[CELEBORN-849][DOC] Document on Master

### PR DESCRIPTION
**Pull request by waitinfuture**
### What changes were proposed in this pull request?
As title.

### Why are the changes needed?
As title.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Manual test.

### CODE REVIEW

**Comment by waitinfuture** [comment]
cc @cfmcgrady @pan3793 @FMX

File: docs/developers/master.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+Round Robin is the simplest allocation strategy. The basic idea is:
+
+- Calculate available slots that can be allocated on each disk
+- Allocate slots among all `Worker`s and all disks in a round-robin fashion, decrement one after allocating, and
+  exclude if no slots available on a disk or `Worker`
+- If the cluster's total available slots is not enough, re-run the algorithm for un-allocated slots as if each
+  disk has infinite capacity
+
+#### Load Aware
+For heterogeneous clusters, `Workers` may have different CPU/disk/network performance, so it's necessary to allocate
```
Comment (pan3793) [inline docs/developers/master.md:94]:
```suggestion
For heterogeneous clusters, `Worker`s may have different CPU/disk/network performance, so it's necessary to allocate
```

File: docs/developers/master.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+Currently, Celeborn supports two allocation strategies:
+
+- Round Robin
+- Load Aware
+
+For both strategies, `Master` will only allocate slots on active `Worker`s with available disks.
+
+During the allocation process, `Master` also simulates the space usage. For example, say a disk's usable space is 1GiB,
+and the estimated file size for each `PartitionLocation` is 64MiB, then at most 16 slots will be allocated on that disk.
```
Comment (pan3793) [inline docs/developers/master.md:82]: could you explain how does Celeborn dynamically adjusts the estimated partition size too?

Comment (waitinfuture) [reply docs/developers/master.md:85]: updated

**Comment by waitinfuture** [comment]
Thanks, merging to main

**Pull request closed without merging**